readetexts-port.patch

```diff
@@ -155,23 +139,81 @@ index b623d59..5de0af0 100644
++                self.books_toolbar.enable_button(True)
+ 
+     def find_books(self, search_text):
+         self.clear_downloaded_bytes()
+@@ -1209,6 +1188,7 @@ class ReadEtextsActivity(activity.Activity):
+         self.list_scroller_visible = True
  
-         self.metadata['activity'] = self.get_bundle_id()
-         self.metadata['mime_type'] = 'application/zip'
-@@ -1146,7 +1130,8 @@ class ReadEtextsActivity(activity.Activity):
-             self.alert(_('Error'), _('You must enter at least one search word.'))
-             self.books_toolbar.search_entry.grab_focus()
-             return
--        f = open('bookcatalog.txt', 'r')
-+        script_dir = os.path.dirname(os.path.realpath(__file__))
-+        f = open(os.path.join(script_dir, 'bookcatalog.txt'), 'r')
-         while f:
-             line = unicode(f.readline(), "iso-8859-1")
-             if not line:
-@@ -1257,30 +1242,13 @@ class ReadEtextsActivity(activity.Activity):
+     def get_book(self):
++        self.retry = 0
+         self.progressbar.show()
+         self.books_toolbar.enable_button(False)
+         self.list_scroller.props.sensitive = False
+@@ -1224,7 +1204,7 @@ class ReadEtextsActivity(activity.Activity):
+                              self.get_old_book_result_cb)
+         else:
+             logger.debug("http://www.gutenberg.org/dirs" +
+-                         self.selected_path + "-8.zip")
++                         self.selected_path + ".zip")
+             GObject.idle_add(self.download_book,  "http://www.gutenberg.org/dirs" + self.selected_path + "-8.zip",
+                              self.get_iso_book_result_cb)
+ 
+@@ -1238,15 +1218,26 @@ class ReadEtextsActivity(activity.Activity):
+         logger.debug("Starting download to %s...", path)
+         try:
+             getter.start(path)
++            self.download_content_length = getter.get_content_length()
++            self.download_content_type = getter.get_content_type()
++            self.textview.grab_focus()
++        except urllib.error.HTTPError as e:
++            self.retry +=1
++            if '.zip' in url and self.retry <= 2:
++                self.download_book("http://www.gutenberg.org/dirs" +
++                               self.selected_path + ".zip",  self.get_book_result_cb)
++            else:
++                self.retry = 0
++                self.alert(_('Error'), _(
++                    'Connection timed out for ') + self.selected_title)
++                self.get_book_error_cb(getter, _("Couldn't find ") + self.selected_title + _('Error') + str(e))
+         except:
+             self.alert(_('Error'), _(
+                 'Connection timed out for ') + self.selected_title)
+-
+-        self.download_content_length = getter.get_content_length()
+-        self.download_content_type = getter.get_content_type()
+-        self.textview.grab_focus()
++            self.get_book_error_cb(getter, _('Connection timed out for ') + self.selected_title)
```

Comment:
Please separate this fix to a separate parch. It will make our life easier in future rebases.